File: module_zhihu/src/main/java/com/gas/zhihu/utils/PagerBeanDbUtils.java
package com.gas.zhihu.utils;

import com.gas.zhihu.bean.PaperBean;
import com.gas.zhihu.db.DaoSession;
import com.gas.zhihu.db.DbUtils;
import com.gas.zhihu.db.PaperBeanDao;

import org.greenrobot.greendao.query.QueryBuilder;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class PagerBeanDbUtils {


    @SuppressWarnings("unchecked")
    public static void insertMapBean(PaperBean bean) {

        DaoSession daoSession = DbUtils.getInstance().getDaoSession();
        PaperBeanDao dao = daoSession.getPaperBeanDao();
        QueryBuilder<PaperBean> qb = dao.queryBuilder();
        List<PaperBean> list =  qb.where(PaperBeanDao.Properties.PathName.eq(bean.getPathName()),
                PaperBeanDao.Properties.VoltageLevel.eq(bean.getVoltageLevel()),
                PaperBeanDao.Properties.MapKey.eq(bean.getMapKey()),
                PaperBeanDao.Properties.FileName.eq(bean.getFileName())).list();
        if (list.size() > 0) {
        } else {
            dao.insert(bean);
        }

    }

    @SuppressWarnings("unchecked")
    public static PaperBean queryData(PaperBean bean) {

        DaoSession daoSession = DbUtils.getInstance().getDaoSession();
        PaperBeanDao dao = daoSession.getPaperBeanDao();
        QueryBuilder qb = dao.queryBuilder();
       return (PaperBean)qb.where(PaperBeanDao.Properties.PathName.eq(bean.getPathName()),
                PaperBeanDao.Properties.VoltageLevel.eq(bean.getVoltageLevel()),
                PaperBeanDao.Properties.MapKey.eq(bean.getMapKey()),
                PaperBeanDao.Properties.FileName.eq(bean.getFileName())).unique();

    }

    @SuppressWarnings("unchecked")
    public static void delete(String path) {
        DaoSession daoSession = DbUtils.getInstance().getDaoSession();
        PaperBeanDao dao = daoSession.getPaperBeanDao();
        QueryBuilder qb = dao.queryBuilder();
        List<PaperBean> list = (ArrayList<PaperBean>) qb.where(PaperBeanDao.Properties.PathName.eq(path)).list();
        if (list != null && !list.isEmpty()) {
            dao.delete(list.get(0));
        }
    }


    @SuppressWarnings("unchecked")
    public static List<PaperBean> queryData(int voltage) {

        DaoSession daoSession = DbUtils.getInstance().getDaoSession();
        PaperBeanDao dao = daoSession.getPaperBeanDao();
        QueryBuilder qb = dao.queryBuilder();
        List<PaperBean> list = (ArrayList<PaperBean>) qb.where(PaperBeanDao.Properties.VoltageLevel.eq(voltage)).list();
        if (list != null && !list.isEmpty()) {
            return list;
        }
        return Collections.emptyList();
    }



    @SuppressWarnings("unchecked")
    public static List<PaperBean> queryAllPaperData() {

        DaoSession daoSession = DbUtils.getInstance().getDaoSession();
        PaperBeanDao dao = daoSession.getPaperBeanDao();
        QueryBuilder qb = dao.queryBuilder();
        List<PaperBean> list = (ArrayList<PaperBean>) qb.list();
        if (list != null && !list.isEmpty()) {
            return list;
        }
        return null;
    }

    @SuppressWarnings("unchecked")
    public static void deleteAll() {

        DaoSession daoSession = DbUtils.getInstance().getDaoSession();
        PaperBeanDao dao = daoSession.getPaperBeanDao();
        if(dao!=null){
            dao.deleteAll();
        }

    }
    @SuppressWarnings("unchecked")
    public static int getDataCount() {

        DaoSession daoSession = DbUtils.getInstance().getDaoSession();
        PaperBeanDao dao = daoSession.getPaperBeanDao();
        QueryBuilder qb = dao.queryBuilder();
        List<PaperBean> list = (ArrayList<PaperBean>) qb.list();
        if (list != null && !list.isEmpty()) {
            return list.size();
        }
        return 0;

    }

    @SuppressWarnings("unchecked")
    public static void updateBean(PaperBean bean) {
        DaoSession daoSession = DbUtils.getInstance().getDaoSession();
        PaperBeanDao dao = daoSession.getPaperBeanDao();
        dao.update(bean);
    }


}

File: module_zhihu/src/main/java/com/gas/zhihu/utils/PagerBeanDbUtils.kt
package com.gas.zhihu.utils

import com.gas.zhihu.bean.PaperBean
import com.gas.zhihu.db.DbUtils
import com.gas.zhihu.db.PaperBeanDao
import org.greenrobot.greendao.query.QueryBuilder
import java.util.*

object PagerBeanDbUtils {
    @JvmStatic
    fun insertMapBean(bean: PaperBean): Boolean {
        val daoSession = DbUtils.getInstance().daoSession
        val dao = daoSession.paperBeanDao
        val qb = dao.queryBuilder()
        val list = qb.where(PaperBeanDao.Properties.PathName.eq(bean.pathName),
                PaperBeanDao.Properties.VoltageLevel.eq(bean.voltageLevel),
                PaperBeanDao.Properties.MapKey.eq(bean.mapKey),
                PaperBeanDao.Properties.FileName.eq(bean.fileName)).list()
        return if (list.size > 0) {
            false
        } else {
            dao.insert(bean)
            true
        }
    }

    @JvmStatic
    fun queryData(bean: PaperBean): PaperBean? {
        val daoSession = DbUtils.getInstance().daoSession
        val dao = daoSession.paperBeanDao
        val qb: QueryBuilder<PaperBean> = dao.queryBuilder()
        val list: List<PaperBean> = qb.where(PaperBeanDao.Properties.PathName.eq(bean.pathName),
                PaperBeanDao.Properties.VoltageLevel.eq(bean.voltageLevel),
                PaperBeanDao.Properties.MapKey.eq(bean.mapKey),
                PaperBeanDao.Properties.FileName.eq(bean.fileName)).list()
        return if (list.isNotEmpty()) {
            list[0]
        } else null

    }

    @JvmStatic
    fun delete(path: String?) {
        val daoSession = DbUtils.getInstance().daoSession
        val dao = daoSession.paperBeanDao
        val qb: QueryBuilder<PaperBean> = dao.queryBuilder()
        val list: List<PaperBean> = qb.where(PaperBeanDao.Properties.PathName.eq(path)).list() as ArrayList<PaperBean>
        if (list.isNotEmpty()) {
            dao.delete(list[0])
        }
    }

    @JvmStatic
    fun queryData(voltage: Int): List<PaperBean> {
        val daoSession = DbUtils.getInstance().daoSession
        val dao = daoSession.paperBeanDao
        val qb = dao.queryBuilder()
        val list: List<PaperBean> = qb.where(PaperBeanDao.Properties.VoltageLevel.eq(voltage)).list() as ArrayList<PaperBean>
        return if (list.isNotEmpty()) {
            list
        } else emptyList()
    }

    @JvmStatic
    fun queryAllPaperData(): List<PaperBean>? {
        val daoSession = DbUtils.getInstance().daoSession
        val dao = daoSession.paperBeanDao
        val qb: QueryBuilder<PaperBean> = dao.queryBuilder()
        val list: List<PaperBean> = qb.list()
        return if (list.isNotEmpty()) {
            list
        } else null
    }

    @JvmStatic
    fun queryAllPaperDataByType(type: Int): List<PaperBean> {
        val daoSession = DbUtils.getInstance().daoSession
        val dao = daoSession.paperBeanDao
        val qb: QueryBuilder<PaperBean> = dao.queryBuilder()
        val list: List<PaperBean> = qb.where(PaperBeanDao.Properties.Type.eq(type)).list()
        return if (list.isNotEmpty()) {
            list
        } else emptyList()
    }

    @JvmStatic
    fun deleteAll() {
        val daoSession = DbUtils.getInstance().daoSession
        val dao = daoSession.paperBeanDao
        dao?.deleteAll()
    }

    @JvmStatic
    val dataCount: Int
        get() {
            val daoSession = DbUtils.getInstance().daoSession
            val dao = daoSession.paperBeanDao
            val qb = dao.queryBuilder()
            val list: List<PaperBean> = qb.list() as ArrayList<PaperBean>
            return if (list.isNotEmpty()) {
                list.size
            } else 0
        }

    @JvmStatic
    fun updateBean(bean: PaperBean) {
        val daoSession = DbUtils.getInstance().daoSession
        val dao = daoSession.paperBeanDao
        dao.update(bean)
    }
}